### PR TITLE
Added apps and tutorials process doc

### PR DESCRIPTION
**Pull request by RangerMauve**
- Adds doc that we collaborated on
- New "tutorials" section where we'll be placing tutorials down the line

We should do a final pass and merge this after everyone's had a look.

Feel free to leave comments as GH reviews.

### CODE REVIEW

File: docs/tutorials/process.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+4: Avoid depending on centralized servers whenever possible.
+Sometimes you mgiht want to pull data from a server for something like an RSS reader, but you should make sure that your app will at least load, or hopefully be useful in some way if that server isn't available.
+
+5: When possible, try using Agregore's built in [theming support](https://agregore.mauve.moe/theming).
+This might involve importing the user-configurable CSS variables from `agregore://theme/vars.css` to add splashes of color, or using the entire `agregore://theme/style.css` file to add default styling to headers and code blocks.
+This is useful to give apps a similar "agregore-y" look and feel and gives users the option to configure the styles of all the apps they use on the local-first web by modifying their Agregore config with custom colors.
+
+These aren't hard restrictions and you're free to break outside of them whenever you see fit, but their main purpose is to keep things simple and to keep the apps local-first as much as we can.
+
```
Comment (dirkcuys) [inline docs/tutorials/process.md:43]: Should we say anything about a license? A permissive license where users don't need to do much to ensure they comply or something more in the spirit of sharing?

File: docs/tutorials/process.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+### Process
+
+ 1. Describe the idea briefly
+ 2. Create at least (at most?) one user story
```
Comment (dirkcuys) [inline docs/tutorials/process.md:48]:
```suggestion
 2. Create at least one user story 
```

File: docs/tutorials/process.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+5: Use the web platform and it's built in APIs.
+This means using stuff like Custom Elements for making reusable components.
+Showing people how to use built in CSS features such as [Sticky Headers](https://developer.mozilla.org/en-US/docs/Web/CSS/position#sticky). There's a lot of cool stuff built into browsers and we can get pretty far by showing people how to use this stuff.
+
+3: Avoid using any build tools like webpack/react/npm/etc.
+Users should be able to install Agregore, and get going on adding code without needing to worry about additional dependencies or build tools.
+Ideally, it should be possible for a person to start off with either Agregore's built in Chromium DevTools, or to use some of the small apps developed as part of these docs.
+
+4: Avoid depending on centralized servers whenever possible.
+Sometimes you mgiht want to pull data from a server for something like an RSS reader, but you should make sure that your app will at least load, or hopefully be useful in some way if that server isn't available.
```
Comment (dirkcuys) [inline docs/tutorials/process.md:36]:
```suggestion
Sometimes you might want to pull data from a server for something like an RSS reader, but you should make sure that your app will at least load, or hopefully be useful in some way if that server isn't available.
```

File: docs/tutorials/process.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+This is useful to give apps a similar "agregore-y" look and feel and gives users the option to configure the styles of all the apps they use on the local-first web by modifying their Agregore config with custom colors.
+
+These aren't hard restrictions and you're free to break outside of them whenever you see fit, but their main purpose is to keep things simple and to keep the apps local-first as much as we can.
+
+
+### Process
+
+ 1. Describe the idea briefly
+ 2. Create at least (at most?) one user story 
+    - e.g. Sammy wants to keep notes on recipes. They go to ipns://cooknot.es to start a notebook. From here they can search for recipes to notate. When they select a recipe a not with the date and a link to the recipe is created
```
Comment (dirkcuys) [inline docs/tutorials/process.md:49]:
```suggestion
    - e.g. Sammy wants to keep notes on recipes. They go to ipns://cooknot.es to start a notebook. From here they can search for recipes to notate. When they select a recipe a note with the date and a link to the recipe is created
```

File: docs/tutorials/process.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
+# Agregore Web Apps and Tutorials Process
+
+## Summary
+
+This document part of a [devgrant](https://github.com/ipfs/devgrants/pull/245) with the [Filecoin Foundation](https://fil.org/) for making demos and tutorials for making and using p2p web apps using [IPFS](https://ipfs.tech/) via [Agregore's](https://agregore.mauve.moe/) p2p protocol handler APIs.
```
Comment (dirkcuys) [inline docs/tutorials/process.md:5]:
```suggestion
This document is part of a [devgrant](https://github.com/ipfs/devgrants/pull/245) with the [Filecoin Foundation](https://fil.org/) for making demos and tutorials for making and using p2p web apps using [IPFS](https://ipfs.tech/) via [Agregore's](https://agregore.mauve.moe/) p2p protocol handler APIs.
```

**Review by dirkcuys — CHANGES_REQUESTED** [review]
Looks good. I left a few small typo corrections and a question about licensing

File: docs/tutorials/process.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+1: Aim to make something that would be useful and or fun at the end.
+Something which a person might want to use beyond the tutorial or that has room for growth for people to tinker with further.
+If it's a note taking app, maybe give them ideas for how to aggregate notes or add custom formatting or text editing functionality.
+Useful doesn't necessarily mean that it has to be utilitarian. Apps that exist to be fun to make and play with are also encouraged since the web is useful tool for people to express themselves.
+
+2: Avoid external dependencies unless absolutely necessary.
+This means reducing your dependency on external CSS and JS libraries like Bootstrap or JQuery. If a library is absolutely necessary (say you're doing stuff with QR code reading and generation), then prefer to show users how to download source files and add them directly to their app's files. This is important to ensure that all the apps we build are fully local first and can load without depending on the cloud. Note that when prompting users to download libraries, you should be specific about which version to use to avoid trouble in the future.
+
+5: Use the web platform and it's built in APIs.
```
Comment (sudocurse) [inline docs/tutorials/process.md:27]: numbering

File: docs/tutorials/process.md (hunk 1 above)
Comment (RangerMauve) [reply docs/tutorials/process.md:43]: I think we're beholden to MIT/Apache 2.0 according to the grant stipulations. I'll add some wording to that extent.

Comment (RangerMauve) [reply docs/tutorials/process.md:43]: Here's the specific wording: https://github.com/ipfs/devgrants/pull/245/files#diff-6ab712642034d99c33fb4d1e1fe1b8ca1d34285baf8a7cb604c7629804a4ad3bR20

**Pull request merged**